metadata/src/main/java/com/redhat/lightblue/rest/metadata/AbstractMetadataResource.java

```diff
@@ -340,18 +288,54 @@ public String removeEntity(@Context SecurityContext sc, @PathParam(PARAM_ENTITY)
     public String clearDefaultVersion(@Context SecurityContext sc, @PathParam(PARAM_ENTITY) String entity) {
         Error.reset();
 
-        final Map<String, List<String>> mappedRoles = metadata.getMappedRoles();
+        try{
+            checkPermission(sc, MetadataRoles.UPDATE_DEFAULTVERSION);
+            return new SetDefaultVersionCommand(null, entity, null).execute();
+        }
+        catch(RoleException e){
+            throw new SecurityException("Unauthorized request for 'DELETE /?entity/default'.", e);
+        }
+    }
+
+    private void checkPermission(SecurityContext sc, MetadataRoles roleAllowed) throws RoleException{
+        final Map<MetadataRoles, List<String>> mappedRoles = metadata.getMappedRoles();
         if(mappedRoles == null || mappedRoles.size() == 0){
             // No authorization was configured
-            return new SetDefaultVersionCommand(null, entity, null).execute();
+            return;
         }
 
-        List<String> roles = mappedRoles.get(MetadataRoles.UPDATE_DEFAULTVERSION.toString());
-        for (String role : roles) {
+        List<String> roles = mappedRoles.get(roleAllowed);
+
+        if(roles.contains(MetadataConstants.ROLE_NOONE)){
+            throw new RoleException(roles);
+        }
+        else if(roles.contains(MetadataConstants.ROLE_ANYONE)){
+            return;
+        }
+
+        for(String role : roles){
             if (sc.isUserInRole(role)){
-                return new SetDefaultVersionCommand(null, entity, null).execute();
+                return;
             }
         }
-        throw new SecurityException("Unauthorized request for 'DELETE /?entity/default'. One of the following roles is needed: "+roles);
+
+        throw new RoleException(roles);
+    }
+
+    private static class RoleException extends Exception{
```

Comment:
Why is this a private inner class?  It ends up being thrown as the cause in `SecurityException`.  Either:
1. make this a proper public class or
2. do not use this class as the cause of any exception thrown by public api's or
3. update `checkPermission` to return a boolean to indicate success or failure and not use exception at all

I recommend the 3rd option.